src/cpp/rtps/reader/StatelessReader.cpp

```diff
@@ -398,65 +398,70 @@ bool StatelessReader::processDataMsg(
     {
         logInfo(RTPS_MSG_IN, IDSTRING "Trying to add change " << change->sequenceNumber << " TO reader: " << m_guid);
 
-        // Ask the pool for a cache change
-        CacheChange_t* change_to_add = nullptr;
-        if (!change_pool_->reserve_cache(change_to_add))
+        // Check rejection by history
+        if (mp_history->can_change_be_added_nts(change, 0) &&
```

Comment:
Same comment than in `StatefulReader`.

Reply:
Fixed on e17467fc61f10f5f4b1c1e38852d72123cf1bd3a